apps/functions-subscription/src/adapters/azure/cosmosdb/__tests__/activation-request.test.ts

```diff
@@ -65,6 +67,48 @@ const makeTestData = (length: number) => {
 };
 
 describe('makeActivationRequestRepository', () => {
+  describe('insert', () => {
+    it('should return the inserted item', async () => {
+      const mockDB = makeDatabaseMock();
+
+      mockDB
+        .container('')
+        .items.create.mockResolvedValueOnce({ resource: anActivationRequest });
+
+      const actual = await makeActivationRequestRepository(
+        mockDB as unknown as Database,
+      ).insert(anActivationRequestWithoutEtag)();
+
+      expect(actual).toStrictEqual(E.right(anActivationRequest));
+      expect(mockDB.container('').items.create).toBeCalledWith(
+        anActivationRequestWithoutEtag,
+      );
+    });
+    it('should return ItemAlreadyExists if the item already exists', async () => {
+      const mockDB = makeDatabaseMock();
+      const error = new ErrorResponse('');
+      // eslint-disable-next-line functional/immutable-data
+      error.code = 409;
+
+      mockDB.container('').items.create.mockRejectedValueOnce(error);
+
+      const actual = await makeActivationRequestRepository(
+        mockDB as unknown as Database,
+      ).insert(anActivationRequestWithoutEtag)();
+
+      expect(actual).toMatchObject(
+        E.left(
+          new ItemAlreadyExists(
+            `The item already exists; original error body: ${error.body}`,
+          ),
+        ),
+      );
+      expect(mockDB.container('').items.create).toBeCalledWith(
+        anActivationRequestWithoutEtag,
+      );
```

Comment:
**suggestion**: What about testing this scenario only in a happy path test case?

```suggestion
```